src/WebJobs.Extensions.DurableTask.Analyzers/SyntaxNodeUtils.cs

```diff
@@ -104,43 +166,11 @@ internal static bool TryGetFunctionAttribute(SyntaxNode attributeExpression, out
             functionAttribute = null;
             return false;
         }
-        
-        public static bool IsMarkedDeterministic(SyntaxNode node)
-        {
-            if (TryGetDeterministicAttribute(node, out SyntaxNode deterministicAttribute))
-            {
-                return true;
-            }
-            return false;
-        }
 
-        public static bool TryGetDeterministicAttribute(SyntaxNode node, out SyntaxNode deterministicAttribute)
-        {
-            if (TryGetMethodDeclaration(node, out SyntaxNode methodDeclaration))
-            {
-                var IEnumeratorAttributeList = methodDeclaration.ChildNodes().Where(x => x.IsKind(SyntaxKind.AttributeList));
-                if (IEnumeratorAttributeList.Any())
-                {
-                    foreach (SyntaxNode attributeList in IEnumeratorAttributeList)
-                    {
-                        if (attributeList.ToString().Equals("[Deterministic]"))
-                        {
-                            deterministicAttribute = attributeList;
-                            return true;
-                        }
-                    }
-                }
-                
-            }
-
-            deterministicAttribute = null;
-            return false;
-        }
-
-        internal static bool TryGetParameterNodeNextToAttribute(AttributeSyntax attributeExpression, SyntaxNodeAnalysisContext context, out SyntaxNode inputType)
+        internal static bool TryGetParameterNodeNextToAttribute(SyntaxNodeAnalysisContext context, AttributeSyntax attributeExpression, out SyntaxNode inputType)
         {
             var parameter = attributeExpression.Parent.Parent;
-            var parameterTypeNamesEnumerable = parameter.ChildNodes().Where(x => x.IsKind(SyntaxKind.IdentifierName) || x.IsKind(SyntaxKind.PredefinedType));
+            var parameterTypeNamesEnumerable = parameter.ChildNodes().Where(x => x.IsKind(SyntaxKind.IdentifierName) || x.IsKind(SyntaxKind.PredefinedType) || x.IsKind(SyntaxKind.GenericName));
             if (parameterTypeNamesEnumerable.Any())
             {
                 inputType = parameterTypeNamesEnumerable.First();
```

Comment:
I see if a few places that you do an `IEnumerable<T>.Any()` statement followed by an `IEnumerable<T>.First()` statement. As a minor performance optimization, consider using `IEnumerable<T>.FirstOrDefault()` and then checking for null. This will avoid evaluating the enumerable multiple times.